jmaqs-webservices-jdk11/src/main/java/com/magenic/jmaqs/webservices/jdk11/WebServiceDriver.java

```diff
@@ -422,6 +422,213 @@ public void setHttpRequestBuilder(HttpRequest.Builder httpRequestBuilder) {
+   *
+   * @param requestUri        The request uri
+   * @param expectedMediaType The type of media being requested
+   * @param content           The put content
+   * @param type              the request type
+   * @param expectedStatus    Assert a success code was returned
+   * @return The response to deserialize as the Http Response
+   * @throws IOException          if the exception is thrown
+   * @throws InterruptedException if the exception is thrown
+   */
+  public <T> T put(String requestUri, MediaType expectedMediaType, Object content,
+      Type type, HttpStatus expectedStatus) throws IOException, InterruptedException {
+    HttpResponse<String> response = this.putContent(requestUri, expectedMediaType, content, expectedStatus);
+    return WebServiceUtilities.deserializeResponse(response, expectedMediaType, type);
+  }
+
+  /**
+   * Execute a web service put.
+   *
+   * @param requestUri        The request uri
+   * @param expectedMediaType The type of media being requested
+   * @param content           The put content
+   * @param expectSuccess     Assert a success code was returned
+   * @return The response body as a string
+   * @throws IOException          if the exception is thrown
+   * @throws InterruptedException if the exception is thrown
+   */
+  public HttpResponse<String> put(String requestUri, MediaType expectedMediaType, Object content,
+      boolean expectSuccess) throws IOException, InterruptedException {
+    return this.putContent(requestUri, expectedMediaType, content, expectSuccess);
+  }
+
+  /**
+   * Execute a web service put.
+   *
+   * @param requestUri        The request uri
+   * @param expectedMediaType The type of media being requested
+   * @param content           The put content
+   * @param expectedStatus     Assert a success code was returned
+   * @return The response body as a string
+   * @throws IOException          if the exception is thrown
+   * @throws InterruptedException if the exception is thrown
+   */
+  public HttpResponse<String> put(String requestUri, MediaType expectedMediaType, Object content,
+      HttpStatus expectedStatus) throws IOException, InterruptedException {
+    return this.putContent(requestUri, expectedMediaType, content, expectedStatus);
+  }
+
+  /**
+   * Execute a web service put.
+   *
+   * @param requestUri        The request uri
+   * @param expectedMediaType The type of media being requested
+   * @param content           The put content
+   * @param postMediaType     The type of the media being put
+   * @param expectSuccess     Assert a success code was returned
+   * @return The response body as a string
+   * @throws IOException          if the exception is thrown
+   * @throws InterruptedException if the exception is thrown
+   */
+  public HttpResponse<String> put(String requestUri, MediaType expectedMediaType, Object content,
+      MediaType postMediaType, boolean expectSuccess) throws IOException, InterruptedException {
```

Comment:
We may need to discuss with the team about the postMediaType parameter stuff. I'm looking at the MAQS version and they have a more advanced or "Custom" way of executing Web Service requests that utilize this parameter (and others). I'm not certain if we should be including that in this PR, as it seems like something we could think about in the future

Reply:
@bradyrad Talked with the team and we decided to get rid of the extra media type